### DISCUSSION TITLE
regrid_data_plane produces -9999 range

### DISCUSSION THREAD

**Discussion by chasecalkins** [dtcenter/METplus#1054] [File I/O] [answered]
I created a netcdf that should have the "internals" of the pcp_combine and I created a 0/1 mask to evaluate hail > 25.4mm. I have been trying to use the regrid_data_plane to regrid onto G211 and I have not been able to produce a value other than -9999. Is there a reason why any option I've tried using while using the regrid_data_plane keeps giving me
**DEBUG 2: Range of regridded data (name="MAX_HAIL_24_mask"; level="(*,*)";) is -9999 to -9999.**

[root@464a5650bbb0 bin]# ./regrid_data_plane /fsx-downscaling/runs/20110426_50L/max_hail_fake_pcp_mask.nc G211 /fsx-downscaling/runs/20110426_50L/max_hail_fake_pcp_mask_211.nc -field 'name="MAX_HAIL_24_mask"; level="(*,*)";' -method MAXGAUSS -width 5 -gaussian_radius 120 -gaussian_dx 81.271 -v 10

DEBUG 7: GaussianInfo::compute() max_r: 5
DEBUG 1: Reading data file: /fsx-downscaling/runs/20110426_50L/max_hail_fake_pcp_mask.nc
DEBUG 4: Met2dDataFileFactory::new_met_2d_data_file() -> created new Met2dDataFile object of type "FileType_NcMet".
DEBUG 4:
DEBUG 4: Lambert Conformal Grid Data:
DEBUG 4:    hemisphere:
DEBUG 4:   scale_lat_1: -9999
DEBUG 4:   scale_lat_2: -9999
DEBUG 4:       lat_pin: -9999
DEBUG 4:       lon_pin: 9999
DEBUG 4:         x_pin: 0
DEBUG 4:         y_pin: 0
DEBUG 4:    lon_orient: 9999
DEBUG 4:          d_km: 4
DEBUG 4:          r_km: 6371
DEBUG 4:            nx: 1199
DEBUG 4:            ny: 799
DEBUG 4:     so2_angle: 0
DEBUG 4:
DEBUG 4: VarInfoFactory::new_var_info() -> created new VarInfo object of type "FileType_NcMet".
DEBUG 2: Input grid: Projection: Lambert Conformal Nx: 1199 Ny: 799 Lat_LL: 81.000 Lon_LL: -81.000 Lon_orient: -81.000 Alpha: 3106.588 Cone: 0.988 Bx: 0.0000 By: 252.2668
DEBUG 3: Use the grid named "G211".
DEBUG 3: Grid Definition: Projection: Lambert Conformal Nx: 93 Ny: 65 Lat_LL: 12.190 Lon_LL: 133.459 Lon_orient: 95.000 Alpha: 203.408 Cone: 0.423 Bx: 51.9999 By: 178.3635
DEBUG 2: Output grid: Projection: Lambert Conformal Nx: 93 Ny: 65 Lat_LL: 12.190 Lon_LL: 133.459 Lon_orient: 95.000 Alpha: 203.408 Cone: 0.423 Bx: 51.9999 By: 178.3635
DEBUG 2: Interpolation options: method = MAXGAUSS, width = 5, shape = SQUARE, vld_thresh = 0.5
DEBUG 7: check_nc_data_2d() count: valid=958001, zero=0, missing=0, 958001 == 958001
DEBUG 7: MetNcFile::data() const -> took 0.01 seconds to read NetCDF data
DEBUG 7: MetNcFile::data() const -> took 0 seconds to fill data plane
DEBUG 4:
DEBUG 4: Data plane information:
DEBUG 4:       plane min: 0
DEBUG 4:       plane max: 1
DEBUG 4:      valid time: 20110426_060000
DEBUG 4:       lead time: 000000
DEBUG 4:       init time: 20110426_060000
DEBUG 4:      accum time: NA
DEBUG 4:
DEBUG 5: interp_gaussian_dp() weight_sum: 0.998469 weight_cnt: 89 nx: 93, ny: 65 max_r: 5
DEBUG 2: Range of input data (name="MAX_HAIL_24_mask"; level="(*,*)";) is 0 to 1.
DEBUG 2: Range of regridded data (name="MAX_HAIL_24_mask"; level="(*,*)";) is -9999 to -9999.
DEBUG 1: Writing output file: /fsx-downscaling/runs/20110426_50L/max_hail_fake_pcp_mask_211.nc

Thanks,
Chase

**Comment by JohnHalleyGotway** [comment, marked as answer]
Chase, I see that you're trying to regrid data...
From this file: /fsx-downscaling/runs/20110426_50L/max_hail_fake_pcp_mask.nc
To: [NCEP Grid 211](https://www.nco.ncep.noaa.gov/pmb/docs/on388/grids/grid211.gif)

The problem is apparent in the log messages you sent. In order to interpolate data to G211, MET needs to know where on earth that data is coming from. So the definition of the source grid must be correct. The bad data values (-9999) in these log messages indicate that it is not:
```
DEBUG 4: Lambert Conformal Grid Data:
DEBUG 4: hemisphere:
DEBUG 4: scale_lat_1: -9999
DEBUG 4: scale_lat_2: -9999
DEBUG 4: lat_pin: -9999
DEBUG 4: lon_pin: 9999
DEBUG 4: x_pin: 0
DEBUG 4: y_pin: 0
DEBUG 4: lon_orient: 9999
DEBUG 4: d_km: 4
DEBUG 4: r_km: 6371
DEBUG 4: nx: 1199
DEBUG 4: ny: 799
DEBUG 4: so2_angle: 0
```
The scale_lat_1, scale_lat_2, lat_pin, lon_pin, and lon_orient values should contain real data values rather than "-9999".

Rather than running regrid_data_plane, I'd recommend that you run plot_data_plane first on the input file. Look at the resulting image to confirm that MET has placed the data at the right spot on earth. That means, make sure the data and background map data line up like they should. After you correct the grid definition in one of those files, please try running:
```
plot_data_plane /fsx-downscaling/runs/20110426_50L/max_hail_fake_pcp_mask.nc max_hail_24_mask.ps 'name="MAX_HAIL_24_mask"; level="(*,*)";'
```
You may find these two sections of the MET User's Guide to be helpful in setting up the grid information:
- https://met.readthedocs.io/en/latest/Users_Guide/appendixB.html?highlight=lambert#grid-specification-strings
- Search for "lambert" in https://met.readthedocs.io/en/latest/Users_Guide/appendixF.html?highlight=lambert#python-embedding-for-2d-data
